packages/ui-calendar/src/Calendar/index.tsx

```diff
@@ -408,17 +496,33 @@ class Calendar extends Component<CalendarProps, CalendarState> {
         days[index].push(day)
         return days // 7xN 2D array of `Day`s
       }, [])
-      .map((row) => (
+      .map((row, rowIndex) => (
         <tr key={`row${row[0].props.date}`} role={role}>
-          {row.map((day, i) => (
-            <td key={day.props.date} role={role}>
-              {role === 'presentation'
-                ? safeCloneElement(day, {
-                    'aria-describedby': this._weekdayHeaderIds[i]
-                  })
-                : day}
-            </td>
-          ))}
+          {row.map((day, i) => {
+            const dayIndex = rowIndex * 7 + i
+            const dayRef = this.state.dayRefs[
+              dayIndex
+            ] as React.LegacyRef<HTMLTableCellElement>
+
+            return (
+              // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
+              <td
+                key={day.props.date}
+                role={role}
+                ref={dayRef}
+                onKeyDown={(e) => this.handleKeyDown(e, dayIndex)}
+              >
+                {role === 'presentation'
+                  ? safeCloneElement(day, {
+                      'aria-describedby': this._weekdayHeaderIds[i],
+                      tabIndex: dayIndex === 0 ? 0 : -1
```

Comment:
check if the onkeydown can be added to the child prop instead of the `td` (if yes, you might need to refactor the code where you call `.firstChild`)

Reply:
also maybe the dayref?